pallets/parachain-staking/src/lib.rs

```diff
@@ -447,13 +451,23 @@ pub mod pallet {
 
 	#[pallet::hooks]
 	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
-		fn on_initialize(n: BlockNumberFor<T>) -> Weight {
+		fn on_initialize(_n: BlockNumberFor<T>) -> Weight {
 			let mut weight = T::WeightInfo::base_on_initialize();
 
+			// Fetch relay slot number
+			let relay_slot = u64::from(T::RelayChainSlotProvider::get());
```

Comment:
@Agusrodri please make it more generic, rename `RelayChainSlotProvider` -> `SlotProvider` and rely on thte slot provider implementation to do the complex maths

Reply:
And basically the `SlotProvider` implementation can provide the para block numlber for moonriver and moonbeam, this way we ensure that we don't brek anything before enabling  async backing